src/ophyd_async/epics/motion/motor.py

```diff
@@ -93,5 +241,62 @@ async def set(
     async def stop(self, success=False):
         self._set_success = success
         # Put with completion will never complete as we are waiting for completion on
-        # the move above, so need to pass wait=False
+        # the move above, so need to pass wait=False.
         await self.motor_stop.trigger(wait=False)
+
+    async def _prepare_velocity(self) -> float:
+        assert self.fly_info
+        fly_velocity = abs(
+            (self.fly_info.start_position - self.fly_info.end_position)
+            / self.fly_info.time_for_move
+        )
+        velocity_max = await self.max_velocity.get_value()
+        if fly_velocity > velocity_max:
+            raise MotorLimitsException(
+                f"Velocity of {fly_velocity}mm/s was requested for a motor with "
+                f"vmax of {velocity_max}mm/s"
+            )
+        await self.velocity.set(fly_velocity)
+        return fly_velocity
+
+    async def _prepare_motor_path(self, fly_velocity: float):
+        assert self.fly_info
+        # Distance required for motor to accelerate to fly_velocity before reaching
+        # start_position, and distance required for motor to decelerate from
+        # fly_velocity to zero after end_position
+        run_up_distance = (await self.acceleration_time.get_value()) * fly_velocity
+
+        self.fly_info.prepared_position = (
+            self.fly_info.start_position - run_up_distance
+            if self.fly_info.direction
+            else self.fly_info.start_position + run_up_distance
+        )
+
+        self.fly_info.completed_position = (
+            self.fly_info.end_position + run_up_distance
+            if self.fly_info.direction
+            else self.fly_info.end_position - run_up_distance
+        )
+
+        motor_lower_limit = await self.low_limit_travel.get_value()
+        motor_upper_limit = await self.high_limit_travel.get_value()
+
+        if (
+            self.fly_info.direction
+            and (
+                self.fly_info.prepared_position < motor_lower_limit
+                or self.fly_info.completed_position > motor_upper_limit
+            )
+        ) or (
+            not self.fly_info.direction
+            and (
+                self.fly_info.prepared_position > motor_lower_limit
+                or self.fly_info.completed_position < motor_upper_limit
+            )
+        ):
```

Comment:
Could: if run_up_distance had the sign of the direction of travel you wouldn't need both sets of if statements below...

Reply:
clever!